workers/unity/Packages/com.improbable.gdk.buildsystem/Configuration/SpatialOSBuildConfigurationEditor.cs

```diff
@@ -10,60 +10,79 @@ namespace Improbable.Gdk.BuildSystem.Configuration
+
         public override void OnInspectorGUI()
         {
             var workerConfiguration = (SpatialOSBuildConfiguration) target;
 
-            EditorGUILayout.BeginHorizontal();
-            workerTypeName = EditorGUILayout.TextField(workerTypeName);
-            var controlRect = EditorGUILayout.GetControlRect(false);
-            var buttonRect = new Rect(controlRect);
-            buttonRect.width = buttonRect.height + 5;
-
-            if (GUI.Button(buttonRect, AddWorkerTypeButtonContents))
+            using (new EditorGUILayout.HorizontalScope())
             {
-                if (workerConfiguration.WorkerBuildConfigurations.All(x => x.WorkerType != workerTypeName))
+                workerTypeName = EditorGUILayout.TextField(workerTypeName);
+
+                GUILayout.FlexibleSpace();
+
+                var canAddWorker = string.IsNullOrEmpty(workerTypeName) ||
+                    workerConfiguration.WorkerBuildConfigurations.Any(c =>
+                        c.WorkerType.Equals(workerTypeName, StringComparison.InvariantCultureIgnoreCase));
+
+                using (new EditorGUI.DisabledScope(canAddWorker))
                 {
-                    var config = new WorkerBuildConfiguration
+                    if (GUILayout.Button(AddWorkerTypeButtonContents, EditorStyles.miniButton))
                     {
-                        WorkerType = workerTypeName,
-                        LocalBuildConfig = new BuildEnvironmentConfig
-                        {
-                            BuildOptions = BuildOptions.Development,
-                        },
-                        CloudBuildConfig = new BuildEnvironmentConfig
+                        EditorUtility.SetDirty(target);
+                        Undo.RecordObject(target, $"Remove '{workerTypeName}'");
+
+                        var config = new WorkerBuildConfiguration
                         {
-                            BuildOptions = BuildOptions.EnableHeadlessMode,
-                        }
-                    };
-                    workerConfiguration.WorkerBuildConfigurations.Add(config);
+                            WorkerType = workerTypeName,
+                            LocalBuildConfig = new BuildEnvironmentConfig
+                            {
+                                BuildOptions = BuildOptions.Development,
+                            },
+                            CloudBuildConfig = new BuildEnvironmentConfig
+                            {
+                                BuildOptions = BuildOptions.EnableHeadlessMode,
+                            }
+                        };
+                        workerConfiguration.WorkerBuildConfigurations.Add(config);
+                    }
                 }
             }
 
-            EditorGUILayout.EndHorizontal();
+            var rect = EditorGUILayout.GetControlRect(false, 1, EditorStyles.foldout);
+            var oldColor = Handles.color;
```

Comment:
This can be replaced with https://docs.unity3d.com/ScriptReference/Handles.DrawingScope-ctor.html

Reply:
Nice, missed that in my search